### PR TITLE
Add app bar to legend and remove sidepanel

### PR DESCRIPTION
**Pull request by NotMyFault**
A minor change to make the 'legend' page consistent with other pages using the app bar, as well as removing the sidepanel.

### Testing done

**Before**:
Screenshot 2022-11-12 at 12 12 31

**After**:
Screenshot 2022-11-12 at 12 11 45

### Proposed changelog entries

- Add breadcrumbs to 'Legend' view and remove the sidepanel.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7365"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by uhafner — CHANGES_REQUESTED** [review]
I'm not sure where we made this design decision (I do not find a reference in our design library) but removing the side panel is in my opinion a bad user experience. The visible view area should not change if there is no reason for it. Maybe for full screen views it makes sense to remove the side panel but not for a simple view that does not require the extra space.

I've seen that we did use this (in my eyes anti-pattern) for other views as well: credentials and  configuration as code are some examples I found. In the side panel area we then show empty space, this does not make much sense. I would suggest that we revert those changes as well. Every view should have a side panel: either a custom one the standard one.

**Comment by janfaracik** [comment]
> I'm not sure where we made this design decision (I do not find a reference in our design library) but removing the side panel is in my opinion a bad user experience. The visible view area should not change if there is no reason for it. Maybe for full screen views it makes sense to remove the side panel but not for a simple view that does not require the extra space.
> 
> I've seen that we did use this (in my eyes anti-pattern) for other views as well: credentials and configuration as code are some examples I found. In the side panel area we then show empty space, this does not make much sense. I would suggest that we revert those changes as well. Every view should have a side panel: either a custom one the standard one.

I'm of the view that pages should only use the side panel if the content of the side panel actually relates to the page itself, e.g. plugin manager, job configuration. In this case having access to 'New item', 'People', etc isn't beneficial to the purpose of the page, which is to explain what the icons mean.

File: core/src/main/resources/jenkins/model/Jenkins/legend.jelly [hunk 1]
```diff
@@ -25,11 +25,11 @@ THE SOFTWARE.
 
 <!-- show the icon legend -->
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:s="/lib/form">
-  <l:layout title="${%Icon legend}">
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson"
+         xmlns:s="/lib/form" xmlns:f="/lib/form">
+  <l:layout title="${%Icon legend}" type="one-column">
     <l:app-bar title="${title}" />
```
Comment (janfaracik) [inline core/src/main/resources/jenkins/model/Jenkins/legend.jelly:31]: LGTM! Would you mind moving the app-bar inside of main-panel, like [this](https://weekly.ci.jenkins.io/design-library/Layouts/).

**Review by janfaracik — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]

File: core/src/main/resources/jenkins/model/Jenkins/legend.jelly (hunk 1 above)
Comment (NotMyFault) [reply core/src/main/resources/jenkins/model/Jenkins/legend.jelly]: For sure

**Comment by uhafner** [comment]
> > I'm not sure where we made this design decision (I do not find a reference in our design library) but removing the side panel is in my opinion a bad user experience. The visible view area should not change if there is no reason for it. Maybe for full screen views it makes sense to remove the side panel but not for a simple view that does not require the extra space.
> > I've seen that we did use this (in my eyes anti-pattern) for other views as well: credentials and configuration as code are some examples I found. In the side panel area we then show empty space, this does not make much sense. I would suggest that we revert those changes as well. Every view should have a side panel: either a custom one the standard one.
> 
> I'm of the view that pages should only use the side panel if the content of the side panel actually relates to the page itself, e.g. plugin manager, job configuration. In this case having access to 'New item', 'People', etc isn't beneficial to the purpose of the page, which is to explain what the icons mean.

I see it differently: for me the side panel is the main navigation bar of Jenkins. Some complex views might replace that menu with a specific menu to reuse that space. But for all other views this navigation bar should not be changed. I do not know of any other application that removes the navigation bar just because you click on one element. You can't even navigate back anymore without using browser controls.

**Comment by Tampa** [comment]
It would make sense if it was a lonely page like the environment variable page opening in its own tab, but seeing as it does not it should retain the navigation to allow going back to the page you came from. The addition to the top bar is fine.

**Comment by timja** [comment]
> It would make sense if it was a lonely page like the environment variable page opening in its own tab, but seeing as it does not it should retain the navigation to allow going back to the page you came from. The addition to the top bar is fine.

> You can't even navigate back anymore without using browser controls.

In the provided screenshot nothing has changed about going back to the page you came from.
to go back you click the dashboard link or use browser back.

**Comment by Tampa** [comment]
The original isn't great design, but arguably this doesn't make it any better. Using the breadcrumbs for navigation might make sense in heavily nested things like builds, but for this it's a bit silly. I'd open this in a new tab, like the environment variables, there you can remove the sidebar or at least present a "go back" button instead of the sidebar. It's a static content you don't interact with, so best practice would be a popup modal.

**Pull request closed without merging**